nbs/18_callback.fp16.ipynb

```diff
@@ -202,18 +202,49 @@
     "    def after_loss(self): self.autocast.__exit__(None, None, None)\n",
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #15.            "Use self as a fake optimizer. self.skipped will be set to True after_step if gradients overflow. "
</pre><p>Q: Is the optimizer step called after this step? or this is the actual optimizer step?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/muellerzr/fastai-docment-sprint/pull/70/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='17'/>

Reply:
The optimizer step is called after before_step. In before_step we fake out the GradScaler in this like: `self.scaler.step(self)`. The GradScaler expects an optimizer here. If we overflow here, then the batch is canceled and the actual Optimizer.step is never called for this batch.

Reply:
OK, so no problem